presets/workspace/inference/vllm/inference_api.py

```diff
@@ -292,6 +292,24 @@ def try_get_max_available_seq_len(args: argparse.Namespace) -> int | None:
         return None
 
 
+def get_max_gpu_memory_utilization() -> float:
+    # Derive an optimal gpu_memory_utilization from the current CUDA free/total
+    # bytes on the active device. This caps vLLM's GPU usage to what is actually
+    # available at runtime (accounting for driver/context/other processes) to reduce
+    # OOM risk during inference. Floor to 2 decimals to avoid rounding
+    # up. See https://github.com/kaito-project/kaito/issues/1374.
+    free_memory, total_memory = torch.cuda.mem_get_info()
```

Comment:
calling this method will initialize the underlying cuda context, which will consume an extra 230MB vram and cannot be released unless process exit.

use instead
```python
def gpu_mem_get_info(device_index: int = 0):
    """Return (free_bytes, total_bytes) for the given GPU device using NVML."""
    pynvml.nvmlInit()
    handle = pynvml.nvmlDeviceGetHandleByIndex(device_index)
    info = pynvml.nvmlDeviceGetMemoryInfo(handle)
    pynvml.nvmlShutdown()
    return info.free, info.total

free_memory, total_memory = gpu_mem_get_info()
```

Reply:
pip install nvidia-ml-py3